compiler/rustc_session/src/config/cfg.rs

```diff
@@ -295,6 +296,9 @@ pub(crate) fn default_configuration(sess: &Session) -> Cfg {
         ins_none!(sym::ub_checks);
     }
 
+    if sess.opts.unstable_opts.emscripten_wasm_eh {
```

Comment:
Furthermore setting should only be set behind nightly. 
```suggestion
    // Nightly-only implementation detail for the `panic_unwind` and `unwind` crate.
    if sess.is_nightly_build() && sess.opts.unstable_opts.emscripten_wasm_eh {
```

Reply:
Isn't this check redundant? The only way for `sess.opts.unstable_opts.emscripten_wasm_eh` to be set is if the user passed `-Z emscripten-wasm-eh` to the cli and the cli parser will error out if any `-Z` flags are passed and it's not a nightly build?

Reply:
Currently yes, but only because `-Zemscripten_wasm_eh` default to `false`, if it's default is changed in the future the cfg will be set in stable which is not what we want. Better be explicit than sorry.

Reply:
Well if we did set `sess.opts.unstable_opts.emscripten_wasm_eh` to `true` but don't emit this `cfg` we'll get crashes in the backend when building the unwind libraries. I suppose that would remind us to come remove this condition?

Reply:
Well, that shouldn't happen, `panic_unwind` and `unwind` are nightly-only and this cfg is internal, it should therefore always at-least be gated under nightly.

Reply:
Okay, I made this change.